### ISSUE TITLE
Python3.8+ no longer needed to link against libpython

### ISSUE THREAD

**Issue by sanjayankur31** [nest/nest-simulator#1630]
**Is your feature request related to a problem? Please describe.**

The Fedora Python maintainers recently informed the community that "As of Python 3.8, python C extensions modules should not link to libpython, unless they embed the interpreter in their code. Relevant upstream PR: https://github.com/python/cpython/pull/12946"

https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/IAIUO5XU54FQM64TDXWF4YMHHWGQFNXT/#6ZJGJ4LHSCDBI4IYAREKXJFENOIVLUNO

As the e-mail also says, a simple check to see whether the tool should link to libpython is to see if the code uses `Py_Initialize` and `Py_Finalize`. I grepped the nest code and didn't find these. So I *think* nest does not need to link to libpython either. 

**Describe the solution you'd like**
Do not link against libpython for python3.8+

**Describe alternatives you've considered**
I could patch the [NeuroFedora NEST package](https://src.fedoraproject.org/rpms/nest) downstream.

**Additional context**
NA.

**Comment by gtrensch** [comment]
@sanjayankur31 Many thanks for pointing this out! Did you went into a NEST issue related to linking? We tend to leave the NEST linkage as it is for the moment, and Python version independent, as long as it does not cause any problems. However, we will keep your hint in mind.

**Comment by sanjayankur31** [comment]
No, I've not observed any issues yet. This was just something that the Python maintainers in Fedora pointed out to all us maintainers. I can carry a patch downstream, but thought it'll be good to point out.

Fedora moves quite quickly, so we're already on Python3.8, and will use Python 3.9 for the next release. So, Nest on NeuroFedora will also use these python versions.

**Comment by brenthuisman** [comment]
Not sure if this is the right place, but since the NeuroFedora [Issues repo](https://github.com/neurofedora/Issues/issues) seems very quiet, I post this also here in the hopes in case someone can use the information.

On a fresh Fedora 32, I ran `sudo dnf install python3-nest`. Starting Python and then `import nest`, I get the following:
```
    [14:17:33 last:0s ~] python
    Python 3.8.5 (default, Aug 12 2020, 00:00:00)
    [GCC 10.2.1 20200723 (Red Hat 10.2.1-1)] on linux
    Type "help", "copyright", "credits" or "license" for more information.
    >>> import nest
    [INFO] [2020.8.25 14:17:38 /builddir/build/BUILD/nest-simulator-2.20.0/nest-simulator-2.20.0/nestkernel/rng_manager.cpp:217 @ Network::create_rngs_] : Creating default RNGs
    [INFO] [2020.8.25 14:17:38 /builddir/build/BUILD/nest-simulator-2.20.0/nest-simulator-2.20.0/nestkernel/rng_manager.cpp:260 @ Network::create_grng_] : Creating new default global RNG

    Aug 25 14:17:38 SLIStartup [Fatal]:
        Your NEST installation seems broken.


    Aug 25 14:17:38 SLIStartup [Fatal]:
        I could not find the startup file that

    Aug 25 14:17:38 SLIStartup [Fatal]:
        should have been in /home/brent/.local/share/nest

    Aug 25 14:17:38 SLIStartup [Fatal]:
        Please re-build NEST and try again.

    Aug 25 14:17:38 SLIStartup [Fatal]:
        The file install.html in NEST's doc directory tells you how.

    Aug 25 14:17:38 SLIStartup [Fatal]:
        Bye.
    >>>
```

I tried the above with also either the `python3-nest-mpich` and `python3-nest-openmpi` packages to no effect.

**Comment by sanjayankur31** [comment]
No, your issue isn't related to this one here. I'm looking at your bug report now (Bugs against neurofedora packages should be reported to the Fedora bugtracker here: https://bugzilla.redhat.com/enter_bug.cgi?product=Fedora&component=nest , similar to how different Linux distributions all have their separate downstream bug-trackers)

**Comment by brenthuisman** [comment]
My problem (see above) was not related to this Issue, I solved it with @sanjayankur31 assistance, see: https://github.com/neurofedora/Issues/issues/1

**Comment by jougs** [comment]
@Helveg: has this been solved in the context of #2195 and #2210? Thanks!

**Comment by Helveg** [comment]
Yep, NEST is now libpython independent :)
